### ISSUE TITLE
Significant drop in accuracy after converting weights.

### ISSUE THREAD

**Issue by blairSmcc03** [TrustworthyComputing/REDsec#5]
Hello,

I am trying to train my own network, following the BYON tutorial. Since the last commit I have been successful however I am noticing a significant drop in accuracy after running `make weight_convert` followed by `make ptxt`. The only changes I have made to the code generated by the compiler is in changing `NUM_SAMPLES` to 100 in main.cpp.

After training for 100 epochs, I get an accuracy of 96.8% in  python. `make ptxt` gives an accuracy around 80%. 

The network I am trying to train is:

![Image](https://github.com/user-attachments/assets/dddcc9dd-6bf9-4d10-bddd-c8d855d0f6f9)

![Image](https://github.com/user-attachments/assets/d0f4d138-478d-4bb5-83e6-4dff69ebabca)

Is this an expected amount of degradation or is it a limitation of this network specifically?  I have experimented will ReLU activations also but found worse results for accuracy, especially in the encrypted domain.

Thanks again for all your help.

Kind Regards,
Blair

**Comment by lfolkerts** [comment]
For sign activations, there should not be any drop in accuracy. This is something we need to try to reproduce and investigate; it is potentially an issue with stride length 2.
For ReLU, there is expected to be a modest drop between the python implementation and plaintext/encrypted domain, due to the discretization of BatchNorm.
